ide/css.visual/src/org/netbeans/modules/css/visual/RuleEditorNode.java

```diff
@@ -595,15 +596,29 @@ private PropertyValuesEditor createPropertyValueEditor(FileObject context, Prope
             GroupGrammarElement rootElement = pmodel.getGrammarElement(context);
 
             rootElement.accept(new GrammarElementVisitor() {
+                private Map<GroupGrammarElement,Void> seen = new IdentityHashMap<>();
+
```

Comment:
It's fine, as in comment above we could also use a map for cleaner semantics.

Reply:
Addressed via https://github.com/apache/netbeans/pull/5432/commits/e2dd93481e809985df55c138448cbaaf0ab75f4b